cs/src/core/Index/FASTER/FASTERThread.cs

```diff
@@ -105,6 +105,7 @@ internal void InitContext(FasterExecutionContext ctx, string token, long lsn = -
                     ctx.retryRequests = new Queue<PendingContext>();
                     ctx.readyResponses = new AsyncQueue<AsyncIOContext<Key, Value>>();
                     ctx.ioPendingRequests = new Dictionary<long, PendingContext>();
+                    ctx.pendingReads = new AsyncCountDown<long, PendingContext>();
```

Comment:
Do we still need this? Why not just use ioPendingRequests as before?

Reply:
All we need now is a counter, no need for a dictionary, correct?

Reply:
Former logic would have ```CompleteIOPendingRequestsAsync``` loop, await and carry the completion of every pending request. Until all pending things are done with.

New logic (thankfully) does not hijack control to do other continuation's business. So, a way to await for the pending count to reach zero is needed.

Also, the counter is a dictionary bcz there's other paths where every element of the pending collection will be traversed:

```csharp
foreach (var _ctx in ctx.pendingReads)
{
    AcquireSharedLatch(_ctx.Value.key.Get());
}
```

Reply:
Also, I foresaw the lock inside ```AsyncCountDown``` would raise concern, so I made it lock free.